src/Uno.UI/DataBinding/BindingPropertyHelper.FastConvert.cs

```diff
@@ -334,6 +339,240 @@ private static bool FastStringConvert(Type outputType, string input, ref object
 			return false;
 		}
 
+		private static bool FastStringToNumericConvert(Type outputType, string input, ref object output)
+		{
+			const NumberStyles numberStyles = NumberStyles.AllowDecimalPoint | NumberStyles.AllowLeadingSign | NumberStyles.Float | NumberStyles.AllowLeadingWhite | NumberStyles.AllowTrailingWhite;
+
+			if (outputType == typeof(double))
+			{
+				if (input == "") // empty string is NaN when bound in XAML
+				{
+					output = double.NaN;
+					return true;
+				}
+
+				if (input.Length == 1)
+				{
+					// Fast path for one digit string-to-double.
+					// Often use in VisualStateManager to set double values (like opacity) to zero or one...
+					switch (input[0])
+					{
+						case '0':
+							output = 0d;
+							return true;
+						case '1':
+							output = 1d;
+							return true;
+						case '2':
+							output = 2d;
+							return true;
+						case '3':
+							output = 3d;
+							return true;
+						case '4':
+							output = 4d;
+							return true;
+						case '5':
+							output = 5d;
+							return true;
+						case '6':
+							output = 6d;
+							return true;
+						case '7':
+							output = 7d;
+							return true;
+						case '8':
+							output = 8d;
+							return true;
+						case '9':
+							output = 9d;
+							return true;
+					}
+				}
+
+				var trimmed = input.Trim();
+
+				if (trimmed.Equals("nan", StringComparison.InvariantCultureIgnoreCase)
```

Comment:
Shouldn't these be case-sensitive?
> In addition to the numeric values, text syntax for x:Double permits the tokens `Infinity`, `-Infinity`, and `NaN`. These tokens are treated as case sensitive.
> https://docs.microsoft.com/en-us/dotnet/desktop/xaml-services/types-for-primitives#xdouble

It seems like the only exception is "Auto": (for wpf, but it seems to be still valid for uwp)
> In XAML you set the value to the string "Auto" (case insensitive) to enable the auto sizing behavior.
> https://docs.microsoft.com/en-us/dotnet/api/system.windows.frameworkelement.width?redirectedfrom=MSDN&view=net-5.0#remarks

Reply:
That's the result of the tests I did on UWP, not the documentation :-)